File: src/lib/java/work/lclpnet/ap2/api/base/MiniGameManager.java
package work.lclpnet.ap2.api.base;

import com.mojang.serialization.Codec;
import net.minecraft.resources.ResourceLocation;
import work.lclpnet.ap2.api.game.MiniGame;

import java.util.Optional;
import java.util.Set;

public interface MiniGameManager {

    Set<MiniGame> getGames();

    Optional<MiniGame> getGame(ResourceLocation gameId);

    Codec<MiniGame> getGameCodec();
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/impl/base/MiniGameManager.kt
package work.lclpnet.ap2.impl.base

import com.mojang.serialization.Codec
import net.minecraft.resources.Identifier
import work.lclpnet.ap2.game.MiniGame

interface MiniGameManager {
    val games: Set<MiniGame>

    fun getGame(gameId: Identifier): MiniGame?

    val gameCodec: Codec<MiniGame>
}
